src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1445,6 +1449,56 @@ export class SavedObjectsRepository {
     };
   }
 
+  async removeReferencesTo(
+    type: string,
+    id: string,
+    options: SavedObjectsRemoveReferencesToOptions = {}
+  ): Promise<SavedObjectsRemoveReferencesToResponse> {
+    const { namespace, refresh } = options;
+    const allTypes = this._registry.getAllTypes().map((t) => t.name);
+
+    // we need to target all SO indices as all types of objects may have references to the given SO.
+    const targetIndices = this.getIndicesForTypes(allTypes);
+
+    const { body } = await this.client.updateByQuery(
+      {
+        index: targetIndices,
+        refresh,
+        body: {
+          script: {
+            source: `
+              if (ctx._source.containsKey('references')) {
+                def items_to_remove = [];
+                for (item in ctx._source.references) {
```

Comment:
An error :) However tbh I wasn't even sure this `if` check really makes any sense in the first place, as we are creating the search dsl with a `hasReference` option set to the value of the ref we want to remove, so `source.references` would likely be assured to be a non empty array here, wdyt?

```ts
 ...getSearchDsl(this._mappings, this._registry, {
        namespaces: namespace ? [namespace] : undefined,
        type: allTypes,
        hasReference: { type, id },
    }),
```

Reply:
That makes sense to me. When I originally wrote this comment, I was concerned about running this script across all `targetIndices`, where we might come across something unexpected. However, since the query is running before this script, we should be reasonably sure that we're dealing with an array.